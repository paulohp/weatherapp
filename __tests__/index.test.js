/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import { Index } from '../pages/index.js'

describe('With Enzyme', () => {
  it('Index shows "Current Location" button', () => {
    const app = shallow(<Index />)
    expect(app.find('p').text()).toEqual('or')
    expect(app.find('#locationButton').text()).toEqual('current position')
  })

  it('Index shows "City" input', () => {
    const app = shallow(<Index />)

    expect(app.find('#cityInput').name()).toEqual('CityInput')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Hello world!"', () => {
    const component = renderer.create(<Index />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})