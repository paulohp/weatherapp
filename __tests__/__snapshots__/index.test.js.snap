// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div
  className="BoxWrapper-sc-1bexchc-0 gbwMGK"
>
  <div
    className="CityWrapper-sc-1suz4et-0 isKHtB"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="CityInput-sc-1xyphig-0 gKkXsI"
        id="cityInput"
        name="city"
        onChange={[Function]}
        placeholder="City"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        <i />
      </button>
    </form>
  </div>
  <p>
    or
  </p>
  <span>
    use my 
  </span>
  <button
    className="LocationButton-rh62ww-0 bdxliu"
    id="locationButton"
    onClick={[Function]}
  >
    current position
  </button>
</div>
`;
